client/README.md

```diff
@@ -1,21 +1,168 @@
-# Zecale client
+# Python client to interact with the Zecale aggregator
 
-## Prerequisites
+## Structure of the directory
 
-- [Python3](https://www.python.org/downloads/) (at least version `3.7`)
-- [Pip](https://pip.pypa.io/en/stable/) (at least version `19.0.2`)
+### `zecale`
+
+```
+zecale
+ |_ api
+ |_ cli
+ |_ core
+ |_ dummy_app
+```
+
+This directory contains the API code for the Zecale client (`api`), its backend implementation (`core`), the code for the client CLI (`cli`), and the code of an helper CLI for a "dummy application" (`dummy_app`).
+
+### `test_commands`
+
+This directory contains a list of useful commands to help run the tests, as well as some minimal testing scenarios acting as integration tests.
+
+### `tests`
+
+The `tests` folder contains the unit tests of the `zecale` package.
 
 ## Setup
 
+Ensure that the following are installed:
+
+- Python 3.7 (See `python --version`)
+- [venv](https://docs.python.org/3/library/venv.html#module-venv) module.
+- gcc
+
+First, make sure to setup the Zeth dependency in order to generate the
+API files and be able to use the Zeth CLIs
 ```console
-$ cd client
+$ cd $ZECALE/depends/zeth/client
 $ python -m venv env
-$ . env/bin/activate
-$ make setup
+$ source env/bin/activate
+(env)$ make setup
+```
+
+Then, Execute the following inside the `client` directory.
+```console
+$ cd $ZECALE/client
+$ python -m venv env
+$ source env/bin/activate
+(env)$ make setup
+```
+
+(It may also be necessary to install solc manually if the `py-solc-x` package
```

Comment:
Should this not be handled by `make setup` in the zeth client?  Did you hit some issue related to this?

Reply:
Ah! Yes, you are right indeed, I'll remove it here. Thanks